### PR TITLE
Add depth camera frame publishing

### PR DESCRIPTION
**Pull request by joshanderson-kw**
Adds Unity script to publish research mode depth camera frames and their associated world matrices.

Replaces the research mode plugin from https://github.com/petergu684/HoloLens2-ResearchMode-Unity with the plugin from https://github.com/microsoft/psi.

Important note: I had issues extracting PV camera and depth camera frames at the same time on the latest HoloLens OS version. Downgrading to version 20348.1432 fixed the issue for me. Probably related to this issue https://github.com/microsoft/HoloLens2ForCV/issues/133﻿.

### CODE REVIEW

File: unity/ARUI/Assets/Scripts/ResearchModeCapture.cs [hunk 1]
```diff
@@ -58,69 +64,159 @@ enum DepthSensorMode
...
-        {
-            TcpServerIPAddr = PTGUtilities.getIPv4AddressString();
-        }
-        catch (InvalidIPConfiguration e)
-        {
-            log.LogInfo(e.ToString());
-            return;
-        }
+#if ENABLE_WINMD_SUPPORT
+        var consent = this.requestCameraAccessTask.Result;
```
Comment (Purg) [inline unity/ARUI/Assets/Scripts/ResearchModeCapture.cs:94]: Is it possible to just access "Result" without assigning it to an unused variable? (which I assume is a build warning)

File: unity/ARUI/Assets/Scripts/ResearchModeCapture.cs [hunk 2]
```diff
@@ -58,69 +64,159 @@ enum DepthSensorMode
...
+            // The rig node may not always be locatable, so we need a null check
+            float[] cameraPose = null;
+            if (rigNodeLocation != null)
+            {
+                // Compute the camera pose from the rig node location
+                cameraPose = this.ToCameraPose(rigNodeLocation);
+            }
+            else
+            {
+                debugString += "rig location is null";
```
Comment (Purg) [inline unity/ARUI/Assets/Scripts/ResearchModeCapture.cs:167]: Should we even output a depth map if we don't have a pose for it?

File: HoloLens2-ResearchMode-Unity/README.md [hunk 3]
```diff
@@ -1,24 +1,29 @@
-# HoloLens2-ResearchMode-Unity
-Unity Plugin for using research mode functionality in HoloLens 2. Modified based on [HoloLens2-ResearchMode-Unity](https://github.com/petergu684/HoloLens2-ResearchMode-Unity).
+# C++/WinRT HoloLens2ResearchMode Project Overview
 
-Skeleton to wrap HoloLens 2 research mode api into Windows Runtime extension. 
+This folder was copied from the microsoft/psi project, found [here](https://github.com/microsoft/psi/tree/master/Sources/MixedReality/HoloLens2ResearchMode).
```
[the remark this answers is not in the document]
Comment (Purg) [reply HoloLens2-ResearchMode-Unity/README.md:3]: Resolved in chat.

File: unity/ARUI/Assets/Scripts/PTGUtilities.cs [hunk 4]
```diff
@@ -55,6 +56,64 @@ public static string getIPv4AddressString()
...
+    {
+        return new float[16] {
+            matrix[0, 0], matrix[0, 1], matrix[0, 2], matrix[0, 3],
+            matrix[1, 0], matrix[1, 1], matrix[1, 2], matrix[1, 3],
+            matrix[2, 0], matrix[2, 1], matrix[2, 2], matrix[2, 3],
+            matrix[3, 0], matrix[3, 1], matrix[3, 2], matrix[3, 3]
+        };
+    }
+
+    public static float[] GetIdentityMatrixFloatArray()
```
[the remark this answers is not in the document]
Comment (joshanderson-kw) [reply unity/ARUI/Assets/Scripts/PTGUtilities.cs:96]: Ok, went ahead and added comment to these functions.

File: unity/ARUI/Assets/Scripts/ResearchModeCapture.cs (hunk 1 above)
Comment (joshanderson-kw) [reply unity/ARUI/Assets/Scripts/ResearchModeCapture.cs:94]: Ooops, I actually meant to have a check to make sure that the camera was accessed OK. It's there now.

File: unity/ARUI/Assets/Scripts/PVCameraCapture.cs [hunk 5]
```diff
@@ -149,14 +165,12 @@ public async Task<bool> InitializeMediaCaptureAsyncTask()
             if (group.DisplayName == "QC Back Camera")
```
[the remark this answers is not in the document]
Comment (joshanderson-kw) [reply unity/ARUI/Assets/Scripts/PVCameraCapture.cs]: Yeah, I added a comment to specify that.

File: unity/ARUI/Assets/Scripts/PVCameraCapture.cs [hunk 6]
```diff
@@ -6,18 +6,17 @@
 using System.IO;
 using System.Linq;
 using System.Net;
+using System.Runtime.InteropServices;
```
[the remark this answers is not in the document]
Comment (joshanderson-kw) [reply unity/ARUI/Assets/Scripts/PVCameraCapture.cs:9]: Haha I think I got that from you.

**Review by Purg — APPROVED** [review]

**Review by Purg — APPROVED** [review]

**Pull request merged**